src/App.tsx

```diff
@@ -1,14 +1,80 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [currentTodos, setCurrentTodos] = useState<Todo[]>([]);
+  const [currentTodo, setCurrentTodo] = useState<Todo | null>(null);
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+  const [filterMode, setFilterMode] = useState('all');
+  const [searchInput, setSearchInput] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const applyFilterAndSearch = (mode: string, searchText: string) => {
+    setFilterMode(mode);
+
+    let filtered = todos;
+
+    if (mode === 'active') {
+      filtered = todos.filter(todo => !todo.completed);
+    } else if (mode === 'completed') {
+      filtered = todos.filter(todo => todo.completed);
+    }
+
+    const finalFiltered = filtered.filter(todo =>
+      todo.title.toLowerCase().includes(searchText.toLowerCase()),
+    );
+
+    setCurrentTodos(finalFiltered);
+  };
```

Comment:
The `applyFilterAndSearch` function is called twice in quick succession in the `filterBy` and `findBy` functions. Consider optimizing this by combining the logic or ensuring that state updates are minimized to avoid unnecessary re-renders.